packages/grid/_modules_/grid/components/cell/GridEditBooleanCell.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import Checkbox, { CheckboxProps } from '@material-ui/core/Checkbox';
+import { GRID_CELL_EDIT_PROPS_CHANGE } from '../../constants/eventsConstants';
+import { GridCellParams } from '../../models/params/gridCellParams';
+
+export function GridEditBooleanCell(props: GridCellParams & CheckboxProps) {
+  const {
+    id: idProp,
+    value,
+    element,
+    formattedValue,
+    api,
+    field,
+    row,
+    colDef,
+    cellMode,
+    getValue,
+    rowIndex,
+    colIndex,
+    isEditable,
+    ...checkboxProps
+  } = props;
+
+  const [valueState, setValueState] = React.useState(value);
+
+  const handleChange = React.useCallback(
+    (event) => {
+      const newValue = event.target.checked;
+      const editProps = { value: newValue };
+      setValueState(newValue);
+      api.publishEvent(GRID_CELL_EDIT_PROPS_CHANGE, { id: idProp, field, props: editProps }, event);
+    },
+    [api, field, idProp],
+  );
+
+  React.useEffect(() => {
+    setValueState(value);
+  }, [value]);
+
+  const id = `MuiDataGrid-cell-${idProp}-${field}`;
+
+  return (
+    <label htmlFor={id} className="MuiDataGrid-editCellBoolean">
```

Comment:
Extra props should be applied on the root (if we want to spread them)

```suggestion
    <label htmlFor={id} className="MuiDataGrid-editCellBoolean" {...other}>
```

Reply:
What's the point of applying the props on the label?

Reply:
We could remove the extra prop. The point was either it's on the root or we don't do it.

Reply:
@m4theushw actually, why do we have a label in the first place?

Reply:
@oliviertassinari The label allows to click in any place of the cell to change the value of the checkbox.